dspot/src/test/java/eu/stamp_project/automaticbuilder/MavenAutomaticBuilderTest.java

```diff
@@ -58,6 +60,48 @@ public void testRunPit() throws Exception {
         assertEquals(15, pitResults.stream().filter(pitResult -> pitResult.getStateOfMutant() == PitResult.State.KILLED).count());
     }
 
+    @Test
+    public void testRunPitDescartes() throws Exception {
+
+        /*
+            Test pit with descartes mutation engine.
+                1- the number of mutants is considerably reduces
+                2- the pom xml is modified before the reset()
+                3- the reset() method restore the original pom.xml
+         */
+
+        Utils.init("src/test/resources/test-projects/test-projects.properties");
+        InputConfiguration.get().setDescartesMode(true);
+        InputConfiguration.get().setFilter("");
+        String pomAsStr = "";
+        try (BufferedReader buffer = new BufferedReader(new FileReader(Utils.getInputConfiguration().getAbsolutePathToProjectRoot() + "pom.xml"))) {
+            pomAsStr = buffer.lines().collect(Collectors.joining(AmplificationHelper.LINE_SEPARATOR));
+        } catch (IOException e) {
+            fail("should not throw the exception " + e.toString());
+        }
+
+        InputConfiguration.get().getBuilder().runPit(Utils.getInputConfiguration().getAbsolutePathToProjectRoot());
+        final List<PitResult> pitResults = PitResultParser.parseAndDelete(Utils.getInputConfiguration().getAbsolutePathToProjectRoot() + Utils.getBuilder().getOutputDirectoryPit());
+
+        try (BufferedReader buffer = new BufferedReader(new FileReader(Utils.getInputConfiguration().getAbsolutePathToProjectRoot() + "pom.xml"))) {
+            assertNotEquals(buffer.lines().collect(Collectors.joining(AmplificationHelper.LINE_SEPARATOR)), pomAsStr);
+        } catch (IOException e) {
+            fail("should not throw the exception " + e.toString());
+        }
+
+        InputConfiguration.get().getBuilder().reset();
+
+        try (BufferedReader buffer = new BufferedReader(new FileReader(Utils.getInputConfiguration().getAbsolutePathToProjectRoot() + "pom.xml"))) {
+            assertEquals(buffer.lines().collect(Collectors.joining(AmplificationHelper.LINE_SEPARATOR)), pomAsStr);
+        } catch (IOException e) {
+            fail("should not throw the exception " + e.toString());
```

Comment:
Suggestion: don't put a catch in a test, it'll bubble up anyway ;) (it's supposed to be an anti pattern).